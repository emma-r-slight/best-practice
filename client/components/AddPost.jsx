import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import { savePost } from '../actions/posts'
import { IfAuthenticated, IfNotAuthenticated } from './Authenticated'

class AddPost extends React.Component {
  state = {
    name: '',
    link: ''
  }

  handleChange = e => {
    const {name, value} = e.target
    this.setState({
      [name]: value
    })
  }

  handleAdd = e => {
    this.props.savePost({
      authorId: this.props.userId,
      ...this.state
    }).then(saved => {
      this.props.history.push(`/post/${saved.id}`)
    })
  }

  render () {
    const { name, link } = this.state
    return (
      <>
        <h2>Add New Post</h2>
        <IfAuthenticated>
          <div>
            <div>Name:</div>
            <input name='name' value={name}
              onChange={this.handleChange} />

            <div>Link:</div>
            <input name='link' value={link}
              onChange={this.handleChange} />

            <button type='button' onClick={this.handleAdd}>Add this post</button>
          </div>
        </IfAuthenticated>
        <IfNotAuthenticated>
          <div>You must <Link to="/signin">sign in</Link> to add new posts.</div>
        </IfNotAuthenticated>
      </>
    )
  }
}

function mapStateToProps (state) {
  return { userId: state.user.id }
}

const mapDispatchToProps = { savePost }

export default connect(mapStateToProps, mapDispatchToProps)(AddPost)
