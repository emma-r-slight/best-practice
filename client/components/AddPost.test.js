import React from 'react'
import { render, screen } from '@testing-library/react'
import { renderWithRedux } from '../testing/utils'
import '@testing-library/jest-dom/extend-expect'
import '@babel/polyfill'

import AddPost from './AddPost'

test('<AddPost> includes name in <li>', async () => {
  const initialEntries = ['/post/2']
  renderWithRedux(<AddPost />)
  const post = await screen.findByTestId('addpost')
  expect(post).toBeInTheDocument()
  expect(post).toMatchSnapshot()
})
