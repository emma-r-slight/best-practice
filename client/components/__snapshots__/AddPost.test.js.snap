// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPost> includes name in <li> 1`] = `
<div
  data-testid="addpost"
>
  <h2>
    Add New Post
  </h2>
  <div>
    You must 
    <a
      href="/signin"
    >
      sign in
    </a>
     to add new posts.
  </div>
</div>
`;
